test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/RemoteAttributeTest.cs

```diff
@@ -518,35 +884,59 @@ public void AddValidation_DoesNotTrounceExistingAttributes()
                 kvp => { Assert.Equal("data-val-remote-url", kvp.Key); Assert.Equal("original", kvp.Value); });
         }
 
-        private static ClientModelValidationContext GetValidationContext(IUrlHelper urlHelper)
```

Comment:
Remaining changes to this class are about rearranging things enough to allow `IModelMetadataProvider` and `IStringLocalizerFactory` overrides. Also made the various `Mock`s strict to ensure only the expected calls occur.